src/components/sidebar/components.test-pw.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useRef } from "react";
+
+import Typography from "../../../src/components/typography";
+import Button from "../button";
+import Sidebar, { SidebarProps } from ".";
+import Box from "../box";
+import Toast from "../toast";
+import Textbox from "../textbox";
+import Dialog from "../dialog";
+
+export const Default = (args: Partial<SidebarProps>) => {
+  const [isOpen, setIsOpen] = useState(true);
+  const onCancel = () => {
+    setIsOpen(false);
+  };
+  return (
+    <>
+      <Button onClick={() => setIsOpen(true)}>Open sidebar</Button>
+      <Sidebar {...args} aria-label="sidebar" open={isOpen} onCancel={onCancel}>
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponent = (props: Partial<SidebarProps>) => {
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open
+        position="right"
+        size="medium"
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponentWithOnCancel = (props: Partial<SidebarProps>) => {
+  const [isOpen, setIsOpen] = useState(true);
+  const handleOnCancel = () => {
+    setIsOpen(false);
+  };
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open={isOpen}
+        position="right"
+        size="medium"
+        onCancel={handleOnCancel}
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarComponentWithHeading = (props: Partial<SidebarProps>) => {
+  return (
+    <>
+      <Sidebar
+        aria-label="sidebar"
+        open
+        position="right"
+        size="medium"
+        header={<Typography variant="h3">Sidebar Header</Typography>}
+        {...props}
+      >
+        <Box mb={2}>
+          <Button buttonType="primary">Test</Button>
+          <Button buttonType="secondary" ml={2}>
+            Last
+          </Button>
+        </Box>
+        <Box mb="3000px">Main content</Box>
+      </Sidebar>
+    </>
+  );
+};
+
+export const SidebarBackgroundScrollTestComponent = () => {
+  return (
+    <Box height="2000px" position="relative">
+      <Box
+        data-element="test-box"
```

Comment:
**suggestion** (non-blocking): I like the idea of adding a `data-element` to identify these boxes in the test 👍 But if we're adding this I think you should remove the `id="bottom-box"` because unless I'm forgetting something, the only reason I added that ID was to locate it in the Cypress test, and we're now locating it by data-element instead. The examples would be cleaner without an un-needed ID.